### DISCUSSION TITLE
Translate languages in "Main language" field in edit a product

### DISCUSSION THREAD

**Discussion by laralem** [openfoodfacts/openfoodfacts-server#6046] [Q&A]
Where is this translation? (English / Portuguese shown in red)

![Edit_OFF](https://user-images.githubusercontent.com/80917261/139940975-038b6310-e80e-45d9-9035-a993777128a6.png)

**Comment by hangy** [comment]
> Where is this translation? (English / Portuguese shown in red)
> 
> ![Edit_OFF](https://user-images.githubusercontent.com/80917261/139940975-038b6310-e80e-45d9-9035-a993777128a6.png)

If I remember correctly, those are taken from https://github.com/openfoodfacts/openfoodfacts-server/blob/bee0f4f0d31376976d24bc17f80bd55d1a2782d3/taxonomies/languages.txt.

**Comment by laralem** [comment]
Thank you. Forgot the search feature in GitHub! Probably the correct branch/file is https://github.com/openfoodfacts/openfoodfacts-server/blob/main/taxonomies/languages.result.txt right?

May be this should be in the translation platform also (or in another like Weblate?). The languages translating "XYZ language" to "language XYZ" have problems. The translation to those languages should be only the name of the language "XYZ" because that way while editing a product we can jump to the first language typing the key. For example, currently in Portuguese if I press the key "L" it jumps to "Latim" but if I press the key "A" for the language "Abecásia" it jumps to another one because "Abecásia" is translated currently to "Língua abecásia".

Unfortunately is not enough and correct, at least in Portuguese, to remove the word "Língua" in "Língua abecásia", because the resulting translation should be "Abecásio" and not "Abecásia" (the gender changes).

Other languages where this seems to happen: 
an:Idioma abkhasio
gl:Lingua abkhaza
la:Lingua Abasca
ro:Limba abhază

It's just a suggestion. Anyway I will fix the Portuguese translations. This "bug" was really annoying me while editing products in OpenFoodFacts website.

**Reply by aleene** [reply]
Sounds like a good correction. So need to change the languages taxonomy. Can put it on my activities list.

**Reply by hangy** [reply]
The languages and their translations were originally sourced from Wikidata, but have been updated a bit now. ONe issue with an external translation tool might be the custom format of the taxonomies, though?

**Reply by laralem** [reply]
If the external translation tool/site (it's Crowdin, didn't remembered) takes more than 15 minutes to implement, may be this was a bad idea. In the end this will be changed rarely. I was thinking too fast. May be this was the reason why this is not in Crowdin from the beginning. Better leave the developers working in other important things.

By the way, in this file is there any reason to have several translations separated by comma like (except the first in English, may be?):
mk:латвиски јазик, летонски јазик
pt:língua letã, idioma letão, letão, linguagem letã, latviešu
ru:латышский язык, латвийский язык

When, at least in the "Main language" field while editing a product, just the first word is used? Key press doesn't work for the first letter of the 2th, 3th... words, just the first word. I don't know if this is used in other situations.
